rosidl_runtime_c/test/benchmark_string_conversion.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "rosidl_runtime_c/string_functions.h"
+#include "rosidl_runtime_c/u16string_functions.h"
+
+#include "performance_test_fixture/performance_test_fixture.hpp"
+
+using performance_test_fixture::PerformanceTest;
+
+BENCHMARK_DEFINE_F(PerformanceTest, string_assign_complexity)(benchmark::State & st)
+{
+  size_t len = st.range(0);
+  std::string data(len, '*');
+
+  rosidl_runtime_c__String s;
+  if (!rosidl_runtime_c__String__init(&s)) {
+    st.SkipWithError("String initialization failed");
+    return;
+  }
+
+  // No resize function - just do a copy
+  rosidl_runtime_c__String__assignn(&s, data.c_str(), len);
+
+  for (auto _ : st) {
+    rosidl_runtime_c__String__assignn(&s, data.c_str(), len);
```

Comment:
Does this need `benchmark::DoNotOptimize` or `benchmark::ClobberMemory`?

Reply:
I tried using that at the beginning, but it didn't seem to make any measurable difference in the performance. Since the function we're calling is already inside of a compiled library, I don't think that `DoNotOptimize` will do very much, but I might try adding the `ClobberMemory` feature to see if that affects the resize case.